Source/SwiftLintFramework/Extensions/Configuration+RulesMode.swift

```diff
@@ -61,8 +61,18 @@ public extension Configuration {
                 self = .only(Set(onlyRules + analyzerRules))
             } else {
                 warnAboutDuplicates(in: disabledRules)
-                warnAboutDuplicates(in: optInRules + analyzerRules)
-                self = .default(disabled: Set(disabledRules), optIn: Set(optInRules + analyzerRules))
+
+                let effectiveOptInRules: [String]
+                if optInRules.contains("all") {
+                    effectiveOptInRules = primaryRuleList.list.compactMap { ruleID, ruleType in
+                        ruleType is OptInRule.Type ? ruleID : nil
+                    }
+                } else {
+                    effectiveOptInRules = optInRules
+                }
+
+                warnAboutDuplicates(in: effectiveOptInRules + analyzerRules)
```

Comment:
What about Analyzer rules? Are they meant to be activated by `all`? If so the "warn about duplicates" should not include them as they have their own section. A warning would be confusing. Admittedly, I think we should not include them.

In case they are skipped the README should mention that.

Reply:
Actually that did occur to me the other day when I saw your PR about analyzer rule handling - I'll check tomorrow and update appropriately

Reply:
So I changed the code to not include analyser rules, but this leads to a weird edge case where the user has listed analyzer rules in `opt_in_rules`, but then `all` would enable all the opt in rules, but disable the explicitly listed analyser rules in the `opt_in_rules` section.

So I went with ignoring the analyser rules, but then taking the union of all the optin rules and the users explicit optin rules, so if the user explicitly listed them, they should remain enabled.

Updated the README as well, but not 100% happy with the language.

Reply:
This semantic makes sense to me. Alternatively, since we have the warning about Analyzer rules listed in the `opt_in_rules` section, we could ignore them in the `all` case. The warning should be enough hint. I'm fine with both ways. Let @jpsim confirm.